### ISSUE TITLE
Image loaded badly (swiped, blurred) in SimpleDraweeView in recyclerview item

### ISSUE THREAD

**Issue by manuel2295** [facebook/fresco#2653]
### Description

I am using frescolib to upload images into recyclerview items. I need to center the images while keeping the aspect ratio. I am using this code to load the image but the images are loading badly, they are streaked, stretched and blurry.

My code for load image:

```
RoundingParams roundingParams = RoundingParams.fromCornersRadius(50f);
        GenericDraweeHierarchy hierarchy =
                GenericDraweeHierarchyBuilder.newInstance(activity.getResources())
                        .setRoundingParams(roundingParams)
                        .setActualImageScaleType(ScalingUtils.ScaleType.CENTER_INSIDE)
                        .setFailureImage(ContextCompat.getDrawable(activity,R.drawable.ic_image_off))
                        .setProgressBarImage(ContextCompat.getDrawable(activity,R.drawable.animview))
                        .build();
        this.imageView.setHierarchy(hierarchy);
        this.imageView.setImageURI(uri);
```

### Reproduction
[this is wrong loaded image](https://i.stack.imgur.com/3VRAz.png)
[this is original image](https://i.stack.imgur.com/ii3Os.jpg)

### Additional Information

* Fresco version: 2.6.0
* Platform version: All devices and android versions that I have tried.

**Comment by oprisnik** [comment]
This is a Fresco limitation when the image does not cover the full viewport that will be fixed with the next Fresco release (2.7.0).

In the meantime, you can either change the scale type or use the overlay color rounding mode instead fi you have a solid background as a workaround for now (which should be possible looking at your screenshot):

RoundingParams roundingParams = RoundingParams.fromCornersRadius(50f);
roundingParams.setOverlayColor(yourBackgroundColorInt);
...
Fresco is using a BitmapShader which repeats the edge pixels that causes this issue. More information: https://frescolib.org/docs/rounded-corners-and-circles.html#caveats

**Comment by manuel2295** [comment]
> This is a Fresco limitation when the image does not cover the full viewport that has been fixed already, which will be included in the next Fresco release (2.7.0).
> 
> In the meantime, you can either change the scale type or use the overlay color rounding mode instead fi you have a solid background as a workaround for now (which should be possible looking at your screenshot):
> ```java
> RoundingParams roundingParams = RoundingParams.fromCornersRadius(50f);
> roundingParams.setOverlayColor(yourBackgroundColorInt);
> ...
> ```
> Fresco is using a BitmapShader which repeats the edge pixels that causes this issue. More information: https://frescolib.org/docs/rounded-corners-and-circles.html#caveats

Thank you for your reply and for the temporary solutions proposed by you. I hope the new version of the library will be released soon.
